### PR TITLE
add polyfill method

### PR DESCRIPTION
**Pull request by sadnessly**
fix the range of lat and lng and add polyfill method

### CODE REVIEW

**Comment by sadnessly** [comment]
It passed with local ide ,which go version is 1.10.8, I don't know why it can't pass in this env.

**Comment by sadnessly** [comment]
> It passed with local ide ,which go version is 1.10.8, I don't know why it can't pass in this env.

**Comment by sadnessly** [comment]
> It passed with local ide ,which go version is 1.10.8, I don't know why it can't pass in this env.

it seems memory calloc has some bugs, I fix it.

**Comment by nrabinowitz** [comment]
Thanks for your contribution. Please note that @joegilley, the primary maintainer of `h3-go`, is out and won't be able to respond to this for several more weeks. Please hold tight until then - thanks!

**Comment by vitaliy-evsyukov** [comment]
Any updates?

File: h3.go [hunk 1]
```diff
@@ -311,9 +311,24 @@ func maxPolyfillSize(geoPolygon *GeoPolygon, res int) int {
...
+		hole_verts_num := C.int(0)
+		holes := new(*C.GeoCoord)
+		return int(C.maxPolyfillSizeGo(numVerts, verts, numHoles, &hole_verts_num, holes, C.int(res)))
+	} else {
+		hole_verts_num := make([]C.int, len(geoPolygon.Holes))
+		holes := make([]*C.GeoCoord, len(geoPolygon.Holes))
+		for i, v := range geoPolygon.Holes {
+			p := (*C.GeoCoord)(C.malloc(C.size_t(C.sizeof_GeoCoord * len(v))))
+			holes[i] = p
+			pa := (*[1 << 30]C.GeoCoord)(unsafe.Pointer(p))
```
Comment (vitaliy-evsyukov) [inline h3.go]: What is "1 << 30" in this case?

**Comment by joegilley** [comment]
thank you for your contributions!  unfortunately we got another pull request for the same feature #19 with a slightly different technique that was more straightforward.   If you can show this technique has substantial performance benefits over #19 using any existing or new benchmarking, I'd like to revisit your alternative.

**Pull request closed without merging**